kedro/logging.py

```diff
@@ -58,3 +58,13 @@ def __init__(self, *args: Any, **kwargs: Any):
             # fixed on their side at some point, but until then we disable it.
             # See https://github.com/Textualize/rich/issues/2455
             rich.traceback.install(**traceback_install_kwargs)  # type: ignore[arg-type]
+
+
+def has_rich_handler(logger: logging.Logger) -> bool:
+    """Returns true if the logger has a RichHandler attached."""
+    return any(isinstance(handler, RichHandler) for handler in logger.handlers)
+
```

Comment:
Is it possible to save this result to avoid calling this hundred times? It's probably have insignificant performance cost.

Reply:
Added `functools.cache`. Side questions: has there been any profiling of Kedro? What are the most expensive parts of the framework?

Reply:
@sbrugman Sorry I missed the notification. I think it depends on what you are targeting.

#1476 is trying to optimise the CLI performance, those are insignificant time for a `kedro run` but significant for CLI UX.

For a `kedro run`, "it depends". The slowest one is always the node computation itself or the dataset I/O. Kedro itself does not have too much penalty, it can be even faster than a script if you can use some of the tricks to speed up (parallel runner, async I/O etc).

The main "penalty" I would say is saving & loading the same data, which could be reduced by using `CacheDataset` or something similar to avoid the unnecessary loading/blocking time.